examples/UpdateContractMint/Program.cs

```diff
@@ -0,0 +1,260 @@
+
+        // Submit the transaction.
+        var txHash = await client.SendAccountTransactionAsync(signedTransaction);
+        Console.WriteLine($"Successfully submitted transfer transaction with hash {txHash}");
+
+        // Watch the status of the transaction, until it becomes finalized in a block.
+        Console.WriteLine($"Waiting for the transaction to finalize...");
+        var finalized = await client.WaitUntilFinalized(txHash);
+
+        Console.WriteLine($"Finalized in block with hash {finalized.BlockHash}");
+
+        // Check whether the transaction was rejected and if so exit.
+        if (finalized.Summary.TryGetRejectedAccountTransaction(out var reason))
+        {
+            Console.WriteLine($"Transaction got rejected:\n{reason}");
+            return;
+        }
+        Console.WriteLine($"Transaction got accepted!");
+
+        // Exact the logs of updated smart contract instances from the outcome.
+        if (!finalized.Summary.TryGetContractUpdateLogs(out var updates))
+        {
+            throw new InvalidOperationException(
+                "Transaction summary failed to parse as a contract update transaction."
+            );
+        }
+        // Print out the events from each updated contract.
+        foreach (var update in updates)
+        {
+            var updatedContract = update.Item1;
+            Console.WriteLine($"Contract {updatedContract} logged:");
+            // If this is our contract, then we use the module schema to parse the events.
+            if (updatedContract == contract)
+            {
+                foreach (var evt in update.Item2)
+                {
+                    // Use the smart contract module schema to deserialize the events.
+                    var json = evt.GetDeserializeEvent(moduleSchema, receiveName.GetContractName());
+                    Console.WriteLine($"- {json}");
+                }
+            }
+            else
+            {
+                // For other contracts we just log the event bytes as a hex encoded string.
+                foreach (var evt in update.Item2)
+                {
+                    Console.WriteLine($"- {evt.ToHexString()}");
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// Class for constructing the JSON representation of the smart contract parameter for 'mint'
+    /// entrypoint of 'cis2_multi' contract.
+    /// The exact structure will depend on the details of the smart contract module.
+    /// </summary>
+    private class MintParameter
+    {
+        /// <summary>
+        /// The owner for the newly minted tokens.
+        ///
+        /// This property only represents the owner being an account address, but the smart contract
+        /// also allow for a contract address here, which we choose not include in the type for
+        /// simplicity.
+        /// </summary>
+        [JsonPropertyName("owner")]
+        public JsonAccountAddr Owner { get; set; }
```

Comment:
I get the simplicity part. 
So this is more of a question. Don't we have "proper" types reflecting these? 

There's quite a lot of invariants on these types, e.g. the address is 32 bytes, the format of the token id etc.

Reply:
We discussed this offline, but for anyone else reading this:
There are proper types defined for the different types, but these do not provide a way to generate the correct JSON. In the future, we might provide this, but not part of this PR.